daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -26,32 +26,36 @@ import scala.collection.immutable.TreeSet
 
 /**
   Speedy builtins are stratified into two layers:
-  Parent: `SBuiltinMaybeHungry`, child: `SBuiltin` (which are never hungry).
-  Hungry means, may throw SpeedyHungry.
+  Parent: `SBuiltinEffect`, child: `SBuiltin` (which are pure).
 
-  Non-hungry builtins can be treated specially because their evaluation is immediate.
+  Effectful builtin functions may raise `SpeedyHungry` exceptions or change machine state.
+  Pure builtins can be treated specially because their evaluation is immediate.
   This fact is used by the execution of the ANF expression form: `SELet1Builtin`.
 
-  The vast majority of the builtins are nevery hungry, and so they extend `SBuiltin`
-  There are 7 hungry builtins which extend `SBuiltinMaybeHungry`
+  Most builtins are pure, and so they extend `SBuiltin`
   */
-/** Speedy builtin functions that may raise `SpeedyHungry` exceptions. */
-private[speedy] sealed abstract class SBuiltinMaybeHungry(val arity: Int) {
+private[speedy] sealed abstract class SBuiltinEffect(val arity: Int) {
   // Helper for constructing expressions applying this builtin.
   // E.g. SBCons(SEVar(1), SEVar(2))
   private[speedy] def apply(args: SExpr*): SExpr =
     SEApp(SEBuiltin(this), args.toArray)
 
   /** Execute the builtin with 'arity' number of arguments in 'args'.
     * Updates the machine state accordingly. */
-  private[speedy] def execute(args: util.ArrayList[SValue], machine: Machine): Unit
+  private[speedy] def executeEffect(args: util.ArrayList[SValue], machine: Machine): Unit
 }
 
-private[speedy] sealed abstract class SBuiltin(val arity1: Int)
-    extends SBuiltinMaybeHungry(arity1) {
-  // TODO: define evaluate, and convert all subclasses to this simpler form
-  // def evaluate(args: util.ArrayList[SValue]): SValue
-  // Then execute can be defined in terms of evaluate. Like how it is done in `SExprAtomic`.
+private[speedy] sealed abstract class SBuiltin(val arity1: Int) extends SBuiltinEffect(arity1) {
+
+  override private[speedy] final def executeEffect(
+      args: util.ArrayList[SValue],
+      machine: Machine): Unit = {
```

Comment:
```suggestion
      machine: Machine,
    ): Unit = {
```

Not a big deal, though. This syntax for multi line arguments helps future diff/merge.

Reply:
Thanks @SamirTalwar-DA 
Should this be the style output by `scalafmt` then?
(Or can that not happen because it requires the programmer to write the trailing `,` ?)

Reply:
We had some issues with IntelliJ making scalafmt output a trailing comma before. This might now be fixed in the latest version of IntelliJ 2019; haven't tested it. I expect we'll need to upgrade to IntelliJ 2020 before we can, though, which means waiting for the Bazel plugin to support it.